test/typescript-karma/Grid/multiColumnSorting.test.js

```diff
@@ -0,0 +1,202 @@
+//@ts-check
+import Grid from '/base/code/grid/es-modules/masters/grid-pro.src.js';
+import U from '/base/code/es-modules/Core/Utilities.js';
+
+const { test } = QUnit;
+const { addEvent } = U;
+
+const waitForAfterSort = (grid) => new Promise((resolve) => {
+    const unbind = addEvent(grid, 'afterSort', () => {
+        unbind();
+        resolve();
+    });
+});
+
+test('Grid multi-column sorting via querying API', async function (assert) {
+    const parentElement = document.getElementById('container');
+    if (!parentElement) {
+        return;
+    }
+
+    const grid = await Grid.grid(parentElement, {
+        dataTable: {
+            columns: {
+                x: [1, 2, 1, 2, 1],
+                y: [5, 4, 3, 2, 1],
+                id: ['a', 'b', 'c', 'd', 'e']
+            }
+        },
+        columns: [{
+            id: 'x'
+        }, {
+            id: 'y'
+        }, {
+            id: 'id'
+        }]
+    }, true);
+    grid.viewport?.resizeObserver?.disconnect();
+
+    await grid.setSorting([
+        { columnId: 'x', order: 'asc' },
+        { columnId: 'y', order: 'asc' }
+    ]);
+
+    assert.deepEqual(
+        grid.viewport?.getColumn('id')?.data,
+        ['e', 'c', 'a', 'd', 'b'],
+        'Grid should be sorted by x (asc) then y (asc)'
+    );
+});
+
+test('Grid multi-column sorting with custom compare', async function (assert) {
+    const parentElement = document.getElementById('container');
+    if (!parentElement) {
+        return;
+    }
+
+    const grid = await Grid.grid(parentElement, {
+        dataTable: {
+            columns: {
+                x: [1, 1, 1, 2, 2],
+                y: [1, 2, 3, 4, 5],
+                id: ['a', 'b', 'c', 'd', 'e']
+            }
+        },
+        columns: [{
+            id: 'x'
+        }, {
```

Comment:
Same here - empty objects (just with ID) that do not define column options are not necessary to be there in my opinion. Check globally.